### ISSUE TITLE
New directive to ignore file

### ISSUE THREAD

**Issue by krufab** [koalaman/shellcheck#1996]
#### For bugs
- Rule Id:
- My shellcheck version: 0.7.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's what I wanted or expected to see:

I think it would be useful to have a new directive to ignore the whole file or part of it.
Or at least a flag when running shellcheck.
I'd see useful when you want to perform the check on *.sh but excluding some files

```bash
#!/usr/bin/env bash
# shellcheck disable=SCXXXX

#This whole file won't be scanned
```

A workaround is:

`shellcheck --external-sources --source-path=SCRIPTDIR --exclude=SC1072,SC1073 installation/scripts/*.sh`

and setting a wrong disable directive:

```bash
#!/usr/bin/env bash
# shellcheck disable=SC*


#This whole file won't be scanned
```

**Comment by ChrisTollefson** [comment]
Good idea!  I agree that this would be useful, especially for dealing with lots of pre-existing/legacy files in a project.  For my part, I like the following syntax:

```sh
#!/usr/bin/env bash
# shellcheck disable=*

echo "This whole file will be ignored."
```

Speaking of that, maybe the ability to similarly ignore all checks for a single line?

```sh
#!/usr/bin/env bash

# shellcheck disable=* # Ignore the following line completely.
echo "Line with lots of errors."
```

### Wishful thinking...

Related to this, perhaps the ability to ignore entire files by adding them to a `.shellcheckignore` file (similar in concept to `.gitignore`)?  Though I realize that's probably more complicated to implement...

Just some ideas.  But anyway, I'm really glad I found shellcheck – it has helped me to improve my scripts dramatically!

**Comment by koalaman** [comment]
In 0.7.2 you can specify a range, e.g. `# shellcheck disable=SC1000-SC9999` and in the next version you'll be able to do `# shellcheck disable=all` 

@docwhat Since ShellCheck currently only processes explicitly listed files without trying to discover them itself, this may already be supported by whichever tool you have that runs ShellCheck (there's currently no way to run `shellcheck foo.sh` and have it *not* check `foo.sh`)
